### PR TITLE
[DEBUG][CI]travis: add more build checks

### PR DESCRIPTION
**Pull request by xiulipan**
Add Werror and Wall flag to check if there are any warnings.
Add sparse check.

Signed-off-by: Pan Xiuli <xiuli.pan@linux.intel.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,17 +16,22 @@ jobs:
         - SHA_PR=`git log --oneline -1 | sed -rn "s/.*Merge (.*) into.*/\1/p"`
         - SHA_MAIN=`git log --oneline -1 | sed -rn "s/.*Merge .* into (.*)/\1/p"`
         - scripts/checkpatch.pl --strict --codespell -g $SHA_MAIN..$SHA_PR
+    - name: "sparse check"
+      script:
+        - export ARCH=x86_64
+        - make defconfig
+        - scripts/kconfig/merge_config.sh .config kconfig/base-defconfig kconfig/sof-defconfig
```
Comment (plbossart) [inline .travis.yml]: maybe add hdaudio-defconfig?

Comment (xiulipan) [reply .travis.yml]: do you mean use this config
`scripts/kconfig/merge_config.sh .config <PATH>/kconfig/base-defconfig <PATH>/kconfig/sof-defconfig <PATH>/kconfig/hdaudio-codecs-defconfig`

Comment (plbossart) [reply .travis.yml]: yep. But just for sparse checks.

**Comment by xiulipan** [comment]
@plbossart 
Update the config for sparse check.

**Pull request merged**